### PR TITLE
Add libraries on mono outerloop build

### LINKED ISSUE #24101: Add libraries outerloop runs running on Mono
We currently only run our outerloop tests on coreclr, we should add mono as well so that outerloop tests are tested on Mono.

cc @dotnet/runtime-infrastructure

### PR DESCRIPTION
**Pull request by safern**
Fixes: https://github.com/dotnet/runtime/issues/24101#issuecomment-685105088
Test Run: https://dnceng.visualstudio.com/public/_build/results?buildId=797679&view=results

This would be adding a mono pipeline named:
`runtime-libraries-mono outerloop`

It would be nice if we could rename the existing ones to contain `-coreclr` to differentiate them easily when looking at the list of available pipelines however I don't know if that would be acceptable outside of an infra rollout window.

cc: @dotnet/runtime-infrastructure

### CODE REVIEW

File: eng/pipelines/libraries/outerloop-mono.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
+trigger: none
+
+schedules:
+- cron: "0 7 * * *" # 7 AM UTC => 12 AM PST
```
Comment (trylek) [inline eng/pipelines/libraries/outerloop-mono.yml:4]: Nit - PST claims to be UTC-8:00 so I think that 7 is actually 11 PM PST.

Comment (safern) [reply eng/pipelines/libraries/outerloop-mono.yml:4]: I see. I got confused by the fact that we're on PDT at the moment and it is `UTC-7` right? https://www.bing.com/search?q=pst+to+utc&cvid=f064666300d74b58948813db38c310e7&FORM=ANAB01&PC=U531

but I think we should leave it as 0 8 and it will be 1AM on PDT and 12AM on PST.

File: eng/pipelines/common/xplat-setup.yml [hunk 2]
```diff
@@ -124,6 +124,9 @@ jobs:
         ${{ if eq(parameters.osGroup, 'tvOS') }}:
           vmImage: 'macOS-10.14'
 
+        ${{ if eq(parameters.osGroup, 'Browser') }}:
+          vmImage: 'ubuntu-latest'
```
Comment (akoeplinger) [inline eng/pipelines/common/xplat-setup.yml:128]: why is this needed?

File: eng/pipelines/libraries/helix-queues-setup.yml [hunk 3]
```diff
@@ -49,26 +49,34 @@ jobs:
...
           - (Debian.10.Amd64.Open)ubuntu.1604.amd64.open@mcr.microsoft.com/dotnet-buildtools/prereqs:debian-10-helix-amd64-bfcd90a-20200121150006
-        - ${{ if eq(parameters.jobParameters.isFullMatrix, false) }}:
-          - Centos.7.Amd64.Open
-          - RedHat.7.Amd64.Open
-          - (Debian.10.Amd64.Open)ubuntu.1604.amd64.open@mcr.microsoft.com/dotnet-buildtools/prereqs:debian-10-helix-amd64-bfcd90a-20200121150006
-          - Ubuntu.1604.Amd64.Open
-          - Ubuntu.1804.Amd64.Open
-          - SLES.15.Amd64.Open
-          - (Fedora.30.Amd64.Open)ubuntu.1604.amd64.open@mcr.microsoft.com/dotnet-buildtools/prereqs:fedora-30-helix-20200512010621-4f8cef7
+        - ${{ if or(ne(parameters.jobParameters.testScope, 'outerloop'), ne(parameters.jobParameters.runtimeFlavor, 'mono')) }}:
```
Comment (akoeplinger) [inline eng/pipelines/libraries/helix-queues-setup.yml:59]: do I parse this yml right that we have a different set of queues for mono vs. coreclr outerloop runs? and that there is no difference between isFullMatrix true/false for mono?

Comment (safern) [reply eng/pipelines/libraries/helix-queues-setup.yml:59]: Yeah I was trying to have mono run on the latest OSs to balance payload and see what people thought about it. If there are any concerns with this I could of course change it.

File: eng/pipelines/common/xplat-setup.yml (hunk 2 above)
Comment (safern) [reply eng/pipelines/common/xplat-setup.yml:128]: This is needed because when you create new pipelines you can't choose a default pipeline anymore other than "Hosted Azure" which forces you to specify a vmImage. In this case Browser was using Ubuntu16.04 hosted for all other pipelines now it will use hosted `ubuntu-latest`

File: eng/pipelines/libraries/outerloop-mono.yml (hunk 1 above)
Comment (safern) [reply eng/pipelines/libraries/outerloop-mono.yml]: Fixed.

**Review by ViktorHofer — APPROVED** [review]
LGTM. Thanks. Did you rename the existing pipeline?

File: src/libraries/sendtohelix.proj [hunk 4]
```diff
@@ -78,6 +78,7 @@
 
   <PropertyGroup>
     <TargetsWindows Condition="'$(TargetOS)' == 'Windows_NT'">true</TargetsWindows>
+    <BuildTargetFramework Condition="'$(TestPackages)' != 'true'">$([MSBuild]::ValueOrDefault('$(BuildTargetFramework)', '$(NetCoreAppCurrent)'))</BuildTargetFramework>
```
Comment (ViktorHofer) [inline src/libraries/sendtohelix.proj:81]: why is this needed?

Comment (ViktorHofer) [reply src/libraries/sendtohelix.proj:81]: Ah because this project isn't invoked via Build.proj which sets the global property. We might want to add a libraries.helix subset so that we can avoid all these translations (we also hardcode the current tfm in some yml files).

Comment (safern) [reply src/libraries/sendtohelix.proj:81]: I'll open an issue, good idea.

**Comment by safern** [comment]
> LGTM. Thanks. Did you rename the existing pipeline?

Not yet. Once I merge I will do it as I need to update the yml file name as I'm renaming it, and send an email to the team.

File: src/libraries/sendtohelix.proj (hunk 4 above)
Comment (ViktorHofer) [reply src/libraries/sendtohelix.proj:81]: Thanks a lot :)

**Comment by safern** [comment]
I've disabled the scheduled builds for `mono-outerloop` until we get them clean; will open an issue to track that effort.

**Comment by safern** [comment]
@danmosemsft in order for me to be able to merge this I will need to backport the yml rename to release/5.0 and release/5.0-rc2 so that we can keep running outerloop tests on those branches; is that OK?

**Comment by danmosemsft** [comment]
This change is to add testing in master right? Can do that in such a way that we can avoid changes in release branches

**Comment by safern** [comment]
Yes we could, if I keep the coreclr outerloop build templates named `outerloop.yml` instead of `outerloop-coreclr.yml` which would be a little bit ambiguous but might not be that bad of a deal. Up to you.

**Comment by danmosemsft** [comment]
Yes please optimize for avoiding release branch churn

**Comment by marek-safar** [comment]
@safern could you please create an issue with tests (assemblies) which fail and need to be fixed

**Comment by safern** [comment]
@marek-safar yeah I will do an analysis and open an issue.

**Comment by bartonjs** [comment]
> This change is to add testing in master right? Can do that in such a way that we can avoid changes in release branches

Can I make a plea for porting the `runtime-libraries outerloop` => `runtime-libraries-coreclr outerloop` rename to the release branch so that we don't immediately have a discrepancy between servicing PRs and master PRs? Anything that makes how two releases build be different makes servicing just that much harder and that much less reliable.

**Comment by safern** [comment]
> Can I make a plea for porting the runtime-libraries outerloop => runtime-libraries-coreclr outerloop rename to the release branch so that we don't immediately have a discrepancy between servicing PRs and master PRs? Anything that makes how two releases build be different makes servicing just that much harder and that much less reliable.

That doesn't depend on the yml... and that should already have taken effect. The pipeline name should be the same for all branches now 😄

**Comment by bartonjs** [comment]
> The pipeline name should be the same for all branches now 😄

My hero! ::swoon::
